src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/coreperson/CorePersonService.kt

```diff
@@ -57,59 +57,69 @@ class CorePersonService(
         },
         sentenceStartDates = allBookings?.flatMap { b -> b.sentences.map { s -> s.startDate } }?.toSortedSet()?.toList() ?: emptyList(),
         nationalities = allBookings?.flatMap { b ->
-          b.profileDetails.filter { pd -> pd.id.profileType.type == "NAT" }.map { n ->
-            OffenderNationality(
-              bookingId = b.bookingId,
-              startDateTime = b.bookingBeginDate,
-              endDateTime = b.getReleaseTime(),
-              latestBooking = b.bookingSequence == 1,
-              nationality = n.profileCode?.toCodeDescription(),
-            )
-          }
+          b.profileDetails.filter { it.id.profileType.type == "NAT" }
+            .filter { it.profileCodeId != null }
+            .map { n ->
+              OffenderNationality(
+                bookingId = b.bookingId,
+                startDateTime = b.bookingBeginDate,
+                endDateTime = b.getReleaseTime(),
+                latestBooking = b.bookingSequence == 1,
+                nationality = n.profileCode!!.toCodeDescription(),
```

Comment:
bit naff the `!!` here, but we've already filtered out the `profileCodeId` null values